### PR TITLE
CNV-18302: Persistent storage support for vTPM

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.14+

Issue: [CNV-18302](https://issues.redhat.com//browse/CNV-18302)

Link to docs preview: https://63333--docspreview.netlify.app/openshift-enterprise/latest/virt/virtual_machines/virt-using-vtpm-devices

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Review by jean-edouard — APPROVED** [review]

File: modules/virt-adding-vtpm-to-vm.adoc [hunk 1]
```diff
@@ -33,9 +36,11 @@ spec:
       domain:
         devices:
           tpm: {} <1>
```
[the remark this answers is not in the document]
Comment (jean-edouard) [reply modules/virt-adding-vtpm-to-vm.adoc:38]: Side note, a robot could have caught that, maybe we should implement a linter for `[source,yaml]` blocks

[the remark this answers is not in the document]
Comment (sjhala-ccs) [reply modules/virt-adding-vtpm-to-vm.adoc:38]: Updated

**Comment by sjhala-ccs** [comment]
@vsibirsk Can you please approve the PR if everything looks good? Thank you.

**Comment by vsibirsk** [comment]
LGTM

**Review by vsibirsk — APPROVED** [review]

File: modules/virt-about-vtpm-devices.adoc [hunk 2]
```diff
@@ -16,6 +16,20 @@ from a Windows 11 image to function without a physical TPM chip.
 If you do not enable vTPM, then the VM does not recognize a TPM device, even if
 the node has one.
 
-vTPM devices also protect virtual machines by temporarily storing secrets
-without physical hardware. However, using vTPM for persistent secret storage is
-not currently supported. vTPM discards stored secrets after a VM shuts down.
+vTPM devices also protect virtual machines by storing secrets without physical hardware. {VirtProductName} supports persisting vTPM device state by using Persistent Volume Claims (PVCs) for VMs. You must specify the storage class to be used by the PVC by setting the `vmStateStorageClass` attribute in the `HyperConverged` custom resource (CR):
```
Comment (kelbrown20) [inline modules/virt-about-vtpm-devices.adoc:19]: Just a nit suggestion. 
```suggestion
A vTPM device also protects virtual machines by storing secrets without physical hardware. {VirtProductName} supports persisting vTPM device state by using Persistent Volume Claims (PVCs) for VMs. You must specify the storage class to be used by the PVC by setting the `vmStateStorageClass` attribute in the `HyperConverged` custom resource (CR):
```

File: modules/virt-adding-vtpm-to-vm.adoc [hunk 3]
```diff
@@ -8,7 +8,10 @@
 
 Adding a virtual Trusted Platform Module (vTPM) device to a virtual machine
 (VM) allows you to run a VM created from a Windows 11 image without a physical
-TPM device. A vTPM device also temporarily stores secrets for that VM.
+TPM device. A vTPM device also stores secrets for that VM.
+
+.Prerequisites
+* You have configured a Persistent Volume Claim (PVC) to use a storage class of type `Filesystem` that supports the `ReadWriteMany` (RWX) access mode. This is necessary for the vTPM device data to persist across VM reboots.
```
Comment (kelbrown20) [inline modules/virt-adding-vtpm-to-vm.adoc:14]:
```suggestion
* You have configured a Persistent Volume Claim (PVC) to use a storage class of type `Filesystem` that supports the `ReadWriteMany` (RWX) access mode. This is necessary for the vTPM device data to persist across VM reboots. 
* You have the OpenShift CLI installed (`oc`)
```

**Review by kelbrown20 — COMMENTED** [review]
Just some small nits, otherwise this looks great!

**Pull request merged**